pkg/datagatherer/versionchecker/versionchecker_test.go

```diff
@@ -0,0 +1,357 @@
+	}
+
+	// ensure there is a valid kubeconfig in a tmp file for the dynamic dg
+	kubeConfig, err := createKubeConfigWithServer(parsedURL.String())
+	if err != nil {
+		t.Fatalf("failed to create temp kubeconfig: %s", err)
+	}
+	defer os.Remove(kubeConfig.Name())
+
+	// ensure there is a valid host file, this would be loaded from a secret
+	// mount in an agent pod
+	hostConfig, err := createDgHostConfigWithServer("http://" + parsedURL.Host)
+	if err != nil {
+		t.Fatalf("failed to create temp kubeconfig: %s", err)
+	}
+	defer os.Remove(hostConfig.Name())
+
+	// create the config for the DataGatherer, wraps config for Dynamic client
+	// and version checker
+	textCfg := fmt.Sprintf(`
+k8s:
+  kubeconfig: %s
+registries:
+- kind: selfhosted
+  params:
+    host: %s
+    bearer: fixtures/example_secret
+`, kubeConfig.Name(), hostConfig.Name())
+
+	config := Config{}
+	err = yaml.Unmarshal([]byte(textCfg), &config)
+	if err != nil {
+		t.Fatalf("failed to load config: %+v", err)
+	}
+
+	dg, err := config.NewDataGatherer(context.Background())
+	if err != nil {
+		t.Fatalf("failed create new dg %s", err)
+	}
+
+	rawResults, err := dg.Fetch()
+	if err != nil {
+		t.Fatalf("failed fetch data %s", err)
+	}
+
+	resultsJSON, err := json.MarshalIndent(rawResults, "", "  ")
+	if err != nil {
+		t.Fatalf("failed to marshal data %s", err)
+	}
+
+	expectedResultsJSON := fmt.Sprintf(`[
+  {
+    "pod": {
+      "kind": "Pod",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "example-6d94489854-zpzhr",
+        "namespace": "example",
+        "selfLink": "/api/v1/namespaces/example/pods/example-6d94489854-zpzhr",
+        "uid": "efff9dae-28ca-42c3-be70-970731c44f67",
+        "resourceVersion": "32023849",
+        "creationTimestamp": null,
+        "labels": {
+          "app": "example"
+        },
+        "ownerReferences": [
+          {
+            "apiVersion": "apps/v1",
+            "kind": "ReplicaSet",
+            "name": "example-6d94489854",
+            "uid": "bb6c0f31-0e28-4c28-a81d-91b8d7bfed33",
+            "controller": true,
+            "blockOwnerDeletion": true
+          }
+        ]
+      },
+      "spec": {
+        "containers": [
+          {
+            "name": "example",
+            "image": "%s/jetstack/example:v1.0.0",
+            "command": [
+              "sh",
+              "-c"
+            ],
+            "resources": {}
+          }
+        ]
+      },
+      "status": {
+        "containerStatuses": [
+          {
+            "name": "example",
+            "state": {},
+            "lastState": {},
+            "ready": false,
+            "restartCount": 0,
+            "image": "",
+            "imageID": "is set"
+          }
+        ]
+      }
+    },
+    "result": {
+      "CurrentVersion": "v1.0.0",
+      "LatestVersion": "v1.0.1",
+      "IsLatest": false,
+      "ImageURL": "%s/jetstack/example"
+    }
+  }
+]`, parsedURL.Host, parsedURL.Host)
```

Comment:
Maybe push the json to a const at the bottom of the file so it doesn't interrupt the flow of the code?

Reply:
Call me mad, but I like this :joy: 

I'm going to leave this comment open in case others agree.